modules/axnet/src/lwip_impl/addr.rs

```diff
@@ -0,0 +1,314 @@
+use core::{
+    fmt::{self, Error},
+    str::FromStr,
+};
+
+use lwip_rust::bindings::{
+    ip4_addr_t, ip_addr_t, lwip_ip_addr_type_IPADDR_TYPE_V4, lwip_ip_addr_type_IPADDR_TYPE_V6,
+};
+
+use core::net::{
+    Ipv4Addr as CoreIpv4Addr, Ipv6Addr as CoreIpv6Addr, SocketAddr as CoreSocketAddr, SocketAddrV4,
+    SocketAddrV6,
+};
+
+/// Mac Address
+#[derive(Clone, Copy, Debug, Default)]
+pub struct MacAddr(pub [u8; 6]);
+
+/// IP Address, either IPv4 or IPv6
+#[derive(Clone, Copy, Debug)]
+pub enum IpAddr {
+    /// IPv4 Address
+    Ipv4(Ipv4Addr),
+
+    /// IPv6 Address
+    Ipv6(Ipv6Addr),
+}
+
+/// IPv4 Address (host byte order)
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Ipv4Addr(pub u32);
+
+/// IPv6 Address
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Ipv6Addr {
+    /// Address in host byte order
+    pub addr: [u32; 4usize],
+
+    /// Zone identifier
+    pub zone: u8,
+}
+
+/// Socket Address (IP Address + Port)
+#[derive(Clone, Copy, Debug)]
+pub struct SocketAddr {
+    /// IP Address
+    pub addr: IpAddr,
+
+    /// Port
+    pub port: u16,
+}
+
+impl MacAddr {
+    /// Create a new MacAddr from a byte array
+    pub fn from_bytes(bytes: &[u8]) -> MacAddr {
+        let mut addr = [0u8; 6];
+        addr.copy_from_slice(bytes);
+        MacAddr(addr)
+    }
+}
+
+impl fmt::Display for MacAddr {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:02x}-{:02x}-{:02x}-{:02x}-{:02x}-{:02x}",
+            self.0[0], self.0[1], self.0[2], self.0[3], self.0[4], self.0[5]
+        )
+    }
+}
+
+impl IpAddr {
+    /// Get the IP Address as a byte array
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            IpAddr::Ipv4(Ipv4Addr(addr)) => unsafe { &*(addr as *const u32 as *const [u8; 4]) },
```

Comment:
I have a question for this conversion: Is big endian or little endian considered here?
There is related method for this conversion in Rust.

Reply:
Ok,I will use to_be_bytes() to make sure Ipaddr is big endian

Reply:
I found  to_be_bytes() returens [u8], not &[u8], so I rewrite Ipv4Addr ,replace  u32 with [u8; 4]